### PR TITLE
feat: new hashing implemented for variants

### PR DESCRIPTION
**Pull request by sjaanus**
1. Variants will be hashed with new seed
2. Fixed a [bug](https://github.com/Unleash/yggdrasil/issues/60), where variant strategy evaluation was not using groupId from strategy

### CODE REVIEW

File: unleash-yggdrasil/src/lib.rs [hunk 1]
```diff
@@ -27,13 +28,15 @@ use unleash_types::client_metrics::{MetricBucket, ToggleStats};
...
+pub const SUPPORTED_SPEC_VERSION: &str = "5.0.2";
+const VARIANT_NORMALIZATION_SEED: u32 = 86028157;
+
 
 pub struct CompiledToggle {
     pub name: String,
     pub enabled: bool,
     pub compiled_strategy: RuleFragment,
-    pub compiled_variant_strategy: Option<Vec<(RuleFragment, Vec<CompiledVariant>)>>,
+    pub compiled_variant_strategy: Option<Vec<(RuleFragment, Vec<CompiledVariant>, String)>>,
```
Comment (sjaanus) [inline unleash-yggdrasil/src/lib.rs:39]: When we are compiling variant strategy, we also need to pull out the groupId from strategy, not only variants.

File: unleash-yggdrasil/src/lib.rs [hunk 2]
```diff
@@ -394,8 +398,8 @@ impl EngineState {
 
     fn resolve_variant<'a>(
         &self,
-        toggle: &CompiledToggle,
```
Comment (sjaanus) [inline unleash-yggdrasil/src/lib.rs:397]: Resolving does not need full toggle, it just needs groupId.

**Review by sighphyre — APPROVED** [review]
LGTM

**Review by gardleopard — APPROVED** [review]
LGTM

**Pull request merged**